ENT-332/ENT-333 Updates Register and Login as per new design

pomegranited opened the pull request:
* Page header: hides Sign in and Register buttons, and updates border
* Reorders form fields, and updates labels, placeholders, and tip text
* Increases clickable area for links and labels.
* Removes "*" markers from required fields, and instead adds "(optional)" labels to optional fields.
* Updates font colors and sizes, and removes box shadows
* Adds custom drop-down and checkbox styles.
* Make Terms of Service and Honor Code open in new window to avoid losing form field values.

**JIRA tickets**: [ENT-332](https://openedx.atlassian.net/browse/ENT-332), [ENT-333](https://openedx.atlassian.net/browse/ENT-333)

**Dependencies**: None

**Screenshots**:

![screen shot 2017-05-03 at 2 10 02 pm](https://cloud.githubusercontent.com/assets/7556571/25648372/403b62e0-300a-11e7-974f-1bea7de91602.png)
![screen shot 2017-05-03 at 2 00 29 pm](https://cloud.githubusercontent.com/assets/7556571/25648238/fe763642-3008-11e7-82c2-d136273a359e.png)

![screen shot 2017-05-03 at 2 10 49 pm](https://cloud.githubusercontent.com/assets/7556571/25648377/5a6c1074-300a-11e7-951d-4f80a2a69d88.png)

![screen shot 2017-05-03 at 2 01 59 pm](https://cloud.githubusercontent.com/assets/7556571/25648249/1ca80a8c-3009-11e7-81b4-497f4e96b2ed.png)

Android Firefox:
![4999841203543302180-account_id 2](https://cloud.githubusercontent.com/assets/7556571/25648261/510b1260-3009-11e7-871a-615948a6f15d.png)

**Sandbox URL**:

* https://pr14985.sandbox.opencraft.hosting/ (71ba54a963)

**Deployment targets**: edx.org and edge.edx.org

**Merge deadline**: 9 May 2017

**Testing instructions**:

1. To see the third-party authentication options, [set up Third Party Authentication](http://edx.readthedocs.io/projects/edx-installing-configuring-and-running/en/latest/configuration/tpa/tpa_integrate_open/tpa_oauth.html) for Google, Facebook, and/or others. 
1. Visit [Register](https://pr14985.sandbox.opencraft.hosting/register), [Sign In](https://pr14985.sandbox.opencraft.hosting/login), and [Forgotten Password]() pages to see the updated design.

**Author notes and concerns**:

1. Not part of the  [ENT-332](https://openedx.atlassian.net/browse/ENT-332) acceptance criteria, but mentioned in the design doc, was the need to make sure that clicking the Terms of Service/Honor Code links didn't result in lost form field values.  Since putting the TOS in a modal dialog is out of scope for this change (and complex, due to white-label alternate site links), I simply made the [links open in a new window](https://github.com/edx/edx-platform/pull/14985/files#diff-701d67b5953c806e8556619039338c96R64). Can revert this change if it's undesirable.

**Reviewers**
- [x] @haikuginger 
- [ ] @brittneyexline 

**Settings**
```yaml
EDXAPP_ENV_EXTRA:
  REGISTRATION_EXTRA_FIELDS: {
    'confirm_email': 'required',
    'level_of_education': 'optional',
    'gender': 'optional',
    'year_of_birth': 'optional',
    'mailing_address': 'optional',
    'goals': 'optional',
    'honor_code': 'required',
    'terms_of_service': 'hidden',
    'city': 'optional',
    'country': 'optional',
  } 
    
EDXAPP_ENABLE_THIRD_PARTY_AUTH: true
EDXAPP_FEATURES:
  PREVENT_CONCURRENT_LOGINS: false
EDXAPP_COMPREHENSIVE_THEME_DIRS:
  - '/edx/app/edxapp/edx-platform/themes/'
EDXAPP_DEFAULT_SITE_THEME: 'edx.org'
EDXAPP_ENABLE_COMPREHENSIVE_THEMING: true
EDXAPP_PLATFORM_NAME: 'edX'
```

edx-webhook commented:
Thanks for the pull request, @pomegranited! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fedx-platform&number=14985"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

haikuginger commented on lms/static/sass/views/_login-register.scss:
```diff
@@ -13,10 +24,24 @@
         margin: 0;
         letter-spacing: normal;
         font-family: $sans-serif;
+        color: #111111;
```
Should this be a fixed color, or use one of the named colors in the edX pattern guide?

pomegranited replied:
Yep, I'm happy to use named colours.  I just wasn't able to find any that matched the design spec exactly, and wasn't sure how strict it was meant to be.

What I found was a [set of tints (lighter) and shades (darker) on `$gray`](https://github.com/edx/edx-platform/blob/master/lms/static/sass/partials/base/_variables.scss#L58-L69):
 
```sass
$gray-l1:   tint($gray,20%)  !default; // #919191 rgb(145,145,145)
$gray-l2:   tint($gray,40%)  !default; // #adadad rgb(173,173,173)
$gray-l3:   tint($gray,60%)  !default; // #c8c8c8 rgb(200,200,200)
                                       // #d9d9d9 rgb(217,217,217)
$gray-l4:   tint($gray,80%)  !default; // #e4e4e4 rgb(228,228,228)
$gray-l5:   tint($gray,90%)  !default; // #f1f1f1 rgb(241,241,241)
$gray-l6:   tint($gray,95%)  !default; // #f8f8f8 rgb(248,248,248)
$gray-l7:   tint($gray,99%)  !default; // #fefefe rgb(254,254,254)
$gray:      rgb(118,118,118) !default; // #767676 rgb(118,118,118)
$gray-d1:   shade($gray,20%) !default; // #5e5e5e rgb(94,94,94)
$gray-d2:   shade($gray,40%) !default; // #474747 rgb(71,71,71)
                                       // #414141 rgb(65,65,65)
$gray-d3:   shade($gray,60%) !default; // #222222 rgb(34,34,34)
$gray-d4:   shade($gray,80%) !default; // #181818 rgb(24,24,24)
                                       // #111111 rgb(17,17,17)
```

Commit 4870d73 uses the closest named colours available to those requested.

haikuginger commented on lms/static/sass/views/_login-register.scss:
```diff
@@ -315,25 +331,69 @@
+        .tip, .label-optional {
+            @extend %t-copy-sub1;
+            margin: 0 0 ($baseline/2) 0;
+            color: #414141;
```
Again, should we be using a named color?

pomegranited replied:
Commit 4870d73 uses the closest named colours available to those requested.

haikuginger commented on lms/static/sass/views/_login-register.scss:
```diff
@@ -3,6 +3,17 @@
 @import '../base/grid-settings';
 @import "neat/neat"; // lib - Neat
 
+body.view-student-account-access {
```
We've actually decided to punt this to ENT-368 so we can think about the implications a bit more, so you can remove this.

pomegranited commented:
@haikuginger Thanks for your review!  I've addressed your comments, updated the sandbox and PR screenshots, and added one to show what I'm seeing in Android now with the new padding.  

The failing bokchoy and lettuce tests are unrelated to this change, so I'm hoping a rebase will fix them.  Let me know when it's ok for me to do that?

I also tested the `edx.org` theme on my devstack by changing these settings in `*.env.json`:
```json
    "COMPREHENSIVE_THEME_DIRS": [
        "/edx/app/edxapp/edx-platform/themes/"
    ],
    "DEFAULT_SITE_THEME": "edx.org",
    "ENABLE_COMPREHENSIVE_THEMING": true,
```

And it looks consistent with the updated styles:
![screen shot 2017-05-01 at 7 21 29 pm](https://cloud.githubusercontent.com/assets/7556571/25577179/78c0d32e-2ea3-11e7-8ba9-0a2e20b0814f.png)
![screen shot 2017-05-01 at 7 21 41 pm](https://cloud.githubusercontent.com/assets/7556571/25577184/7b150f64-2ea3-11e7-8171-408f80e60d84.png)

Let me know if I should enable these settings on  the sandbox?

haikuginger commented:
@pomegranited, go ahead and rebase on master - and yes, please enable the edx.org theme on your sandbox.

pomegranited commented:
Thanks @haikuginger , I'm spinning up a new sandbox server now.

douglashall commented:
@pomegranited @haikuginger I'm seeing some discrepancies between the sandbox and the mockup in the Jira ticket:

- [ ] Sign In H1 is below the "First time here?" text.
- [ ] Sign In button does not extend the full width of the form.
![screen shot 2017-05-01 at 1 58 04 pm](https://cloud.githubusercontent.com/assets/224634/25588445/49243aa2-2e76-11e7-83bc-7da281f1dc77.png)

pomegranited commented:
Hi @douglashall  There was some further discussion on the invision mockups that went beyond the initial redline snapshots.  Please see these pages: [Register](https://projects.invisionapp.com/share/JWBDY0TP6#/screens/229971836) and [Sign In](https://projects.invisionapp.com/share/JWBDY0TP6#/screens/229971826).

douglashall commented:
@pomegranited Thanks for the clarification!

brittneyexline commented:
@pomegranited if it's not too much trouble, could we update the configuration on your sandbox to have the same fields visible as edx.org and to change the platform name to edX? It will make it easier for Griff and Christine to review.

pomegranited commented:
@brittneyexline Sure thing!  I've updated the sandbox configuration to show just the `honor` checkbox, and hide the `terms_of_service` one, which combines the text together.  Also changed the platform name to `edX`.

* ddfb637 adjusts the checkbox to fix the outline border
* 4aa1c7a, 513889a, e5993bd, 6f946ac address Christine's review points from ENT-332

Could you please have another look, and let me know if this is ready to squash and merge?

brittneyexline commented:
@pomegranited, there are a few fixes Christine would like to see before she signs off - they are detailed in the comments on the jira tickets. After those are addressed you should be good to go and I'll formally give my thumbs up.

pomegranited commented:
Thanks, @brittneyexline .  I've updated the height of the drop-down boxes, and the width of the form is addressed by https://github.com/edx/edx-platform/pull/15025.

The only remaining issue is one I'm having trouble reproducing:
> page jump (horizontal) when I go between Sign In and Register.

Are you seeing this, and on which browser/OS?  I did add [some JS](https://github.com/edx/edx-platform/pull/14985/files#diff-aa10400ca417043aab41f33394c90790R238) to reset the scrolling after the `form.focus()` changes, since that was scrolling the page to the top of the form, so maybe there's a judder?  I believe the `form.focus()` line needs to stay though, for accessibility.  Any ideas are welcome.

pomegranited commented:
jenkins run js

haikuginger commented:
@brittneyexline, as a heads-up, @pomegranited is taking some time off, so I'll be taking care of getting this across the line.

haikuginger commented:
Rebased from 13a8ec93c9.

haikuginger commented:
Sandbox updated to bdd8e5d.

bradenmacdonald commented on lms/static/sass/views/_login-register.scss:
```diff
@@ -10,13 +10,34 @@
+
+    h3.fields-required {
+        font-size: 14px;
+        color: #414141;
```
This should be a variable from http://ux.edx.org/design_elements/colors/ not a hex code.
